atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2509,38 +2563,80 @@ private void logCommitLockTenureExceeded(
                 stackTraceSnapshot);
     }
 
-    protected Set<LockDescriptor> getLocksForWrites() {
-        Set<LockDescriptor> result = new HashSet<>();
+    protected LocksAndMetadata getLocksAndMetadataForWrites() {
+        Set<LockDescriptor> lockDescriptors = new HashSet<>();
```

Comment:
I think this makes sense to track these separately, due to the fact that otherwise we would need to do multiple filters of a map otherwise at the end, which is a bit ugly